config/configload/merge.go

```diff
@@ -547,24 +547,24 @@ func mergeDefinitions(bodies []*hclsyntax.Body) (*hclsyntax.Block, map[string]*h
 
 func mergeDefaults(bodies []*hclsyntax.Body) (*hclsyntax.Block, error) {
 	attrs := make(hclsyntax.Attributes)
-	envVars := make(map[string]cty.Value)
+	envVars := make(map[string]hcl.Expression)
 
 	for _, body := range bodies {
 		for _, block := range body.Blocks {
 			if block.Type == defaults {
 				for name, attr := range block.Body.Attributes {
 					if name == environmentVars {
-						v, err := eval.Value(nil, attr.Expr)
-						if err != nil {
-							return nil, err
+						expObj, ok := attr.Expr.(*hclsyntax.ObjectConsExpr)
+						if !ok {
+							r := attr.Expr.Range()
+							return nil, newDiagErr(&r, fmt.Sprintf("error: %s must be object type", environmentVars))
```

Comment:
```suggestion
							return nil, newDiagErr(&r, fmt.Sprintf("%s must be object type", environmentVars))
```
"error: " not necessary, IMO